packages/react/src/components/Popover/index.js

```diff
@@ -23,20 +32,142 @@ const Popover = React.forwardRef(function Popover(props, ref) {
     ...rest
   } = props;
   const prefix = usePrefix();
+  const floating = useRef();
+  const popover = useRef();
+  const ref = useMergedRefs([forwardRef, popover]);
+  const [autoAligned, setAutoAligned] = useState(false);
+  const [autoAlignment, setAutoAlignment] = useState(align);
   const className = cx({
     [`${prefix}--popover-container`]: true,
     [`${prefix}--popover--caret`]: caret,
     [`${prefix}--popover--drop-shadow`]: dropShadow,
     [`${prefix}--popover--high-contrast`]: highContrast,
     [`${prefix}--popover--open`]: open,
-    [`${prefix}--popover--${align}`]: true,
+    [`${prefix}--popover--${autoAlignment}`]: autoAligned,
+    [`${prefix}--popover--${align}`]: !autoAligned,
     [customClassName]: !!customClassName,
   });
 
+  useIsomorphicEffect(() => {
+    if (!autoAlign) {
+      setAutoAligned(false);
+      return;
+    }
+
+    if (!floating.current) {
+      return;
+    }
+
+    if (autoAligned === true) {
+      return;
+    }
+
+    const rect = floating.current.getBoundingClientRect();
+
+    // The conditions, per side, of when the popover is not visible, excluding the popover's internal padding(16)
```

Comment:
Will the popover always have this internal padding? I had thought this was specific to the story and wanted to check if that padding changed if this still would work